src/model/transaction/applyEntityToContentState.ts

```diff
@@ -16,19 +16,22 @@ import {
   SelectionState,
 } from '../immutable/SelectionState';
 import {flatten, map, skipUntil, takeUntil} from '../descript/Iterables';
-import applyEntityToContentBlock from './applyEntityToContentBlock';
+import applyEntityToContentBlock, {
+  applyEntityToMutableCharacterList,
+} from './applyEntityToContentBlock';
 import {
   makeBlockMapIndex,
   BlockMapIndex,
   mergeMapUpdates,
   BlockMap,
 } from '../immutable/BlockMap';
 import {BlockNode} from '../immutable/BlockNode';
+import {CharacterMetadata} from '../immutable/CharacterMetadata';
 
 function updateNewBlocksWithEntity(
   blockMap: BlockMap,
   blockMapIndex: BlockMapIndex,
-  newBlocks: Record<string, BlockNode>,
+  newCharacterLists: Map<string, CharacterMetadata[]>,
```

Comment:
add a doc on what `string` is in this case (block keys?). maybe rename the param to newBlockCharacterLists